core/ordersync/ordersync.go

```diff
@@ -449,68 +445,88 @@ func (s *Service) getOrdersFromPeer(ctx context.Context, providerID peer.ID) err
 		_ = stream.Close()
 	}()
 
-	var nextReq *Request
-	var selectedSubprotocol Subprotocol
+	// First request
+	nextReq := &rawRequest{
+		Type:         TypeRequest,
+		Subprotocols: s.SupportedSubprotocols(),
+		Metadata:     nil,
+	}
+	nextReq, err = s.makeOrderSyncRequest(ctx, nextReq, stream, providerID)
+	if err != nil {
+		return err
+	}
+	if nextReq == nil {
+		return errors.New("received 0 orders from peer")
+	}
+
 	for {
 		select {
 		case <-ctx.Done():
 			return ctx.Err()
 		default:
 		}
-
-		var rawReq *rawRequest
+		nextReq, err = s.makeOrderSyncRequest(ctx, nextReq, stream, providerID)
```

Comment:
The diff makes this change hard to review, but this function is much cleaner now. Any naming suggestions would be welcome, as `makeOrderSyncRequest` doesn't feel quite right (although it could be worse)

Reply:
I think this is fine. The way the interface is defined, I don't think there's a great name for what the function has to do.